### PR TITLE
BAU: cleanup config paths

### PR DESCRIPTION
**Pull request by gidsg**
### Change description
Attempt to address PR comment: https://github.com/communitiesuk/funding-service-design-authenticator/pull/346#discussion_r1824135986 replaces a bunch of instances of paths defined in config, so only the hostname should be defined.

(There is probably more to do on this but wanted to raise a PR before this got too big to review)

- [ ] Unit tests and other appropriate tests added or updated
- [ ] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
This is a refactor so tests should still pass and app should work as before

### CODE REVIEW

**Review by hamzabinkhalid — DISMISSED** [review]

File: config/envs/default.py [hunk 1]
```diff
@@ -153,10 +140,6 @@ class DefaultConfig(object):
     """
     MAGIC_LINK_EXPIRY_DAYS = 1
     MAGIC_LINK_EXPIRY_SECONDS = 86400 * MAGIC_LINK_EXPIRY_DAYS
-    if APPLICANT_FRONTEND_HOST:
-        MAGIC_LINK_REDIRECT_URL = APPLICANT_FRONTEND_HOST + "/account"
-    else:
-        MAGIC_LINK_REDIRECT_URL = "https://www.gov.uk/error"
```
Comment (gidsg) [inline config/envs/default.py:159]: This logic is not needed as the value is defaulted so always present.

File: api/sso/routes.py [hunk 2]
```diff
@@ -115,7 +115,9 @@ def get_token(self):
             roles=session["user"].get("roles") or [],
         )
 
-        redirect_url = Config.ASSESSMENT_POST_LOGIN_URL  # TODO: Remove defaulting to Assessment, instead use return_app
+        redirect_url = (  # TODO: Remove defaulting to Assessment, instead use return_app
+            f"{Config.ASSESSMENT_FRONTEND_HOST}/assess/fund_dashboard"
```
Comment (sfount) [inline api/sso/routes.py:119]:
```python
from urllib.parse import urljoin
```

```suggestion
            ulrjoin(Config.ASSESSMENT_FRONTEND_HOST, "/assess/fund_dashboard")
```

I think I have a small preference for using the core url libs as they'll handle some edge cases around trailing `/` and things but I don't think it's blocking.

(Also applies to most the changes)

Removing things that don't change based on environment from the config feels like a clearer pattern to follow, great stuff!

File: frontend/magic_links/routes.py [hunk 3]
```diff
@@ -84,10 +84,7 @@ def landing(link_id):
         app_guidance = None
         if round_data.application_guidance:
             app_guidance = round_data.application_guidance.format(
-                all_questions_url=Config.APPLICATION_ALL_QUESTIONS_URL.format(
-                    fund_short_name=fund_short_name,
-                    round_short_name=round_short_name,
-                )
+                all_questions_url=f"{Config.APPLICANT_FRONTEND_HOST}/all_questions/{fund_short_name}/{round_short_name}"
```
Comment (gidsg) [inline frontend/magic_links/routes.py:87]: Very confused about what this code is doing as this variable appears to be unused, so leaving alone for now.

File: models/account.py [hunk 4]
```diff
@@ -70,7 +70,7 @@ def get_account(email: str = None, account_id: str = None, azure_ad_subject_id=N
         if email is account_id is azure_ad_subject_id is None:
             raise TypeError("Requires an email address, azure_ad_subject_id or account_id")
 
-        url = Config.ACCOUNT_STORE_API_HOST + Config.ACCOUNTS_ENDPOINT
+        url = Config.ACCOUNT_STORE_API_HOST + "/accounts"
```
Comment (gidsg) [inline models/account.py:73]: Wanted to update this but lots of the unit tests break as they depend on the invalid url `account_store` and appears non-trivial to fix.

**Comment by gidsg** [comment]
I think there is more we can do here especially around resolving absolute URLs using `SERVER_ROOT` but that probably belongs in a separate PR.

**Review by sfount — APPROVED** [review]
Looks like a great clean up, removing things that don't change form the environment config makes it much easier to read whats important in there at a glance.

I'm approving from a code change point of view, I don't feel like I know what some of the URLs + especially the fund/ round argument specific behaviour does but hopefully someone else can speak to that

File: app.py [hunk 5]
```diff
@@ -114,17 +116,15 @@ def before_request_modifier():
...
     @flask_app.context_processor
     def inject_global_constants():
+        query_params = urlencode({"fund": request.args.get("fund", ""), "round": request.args.get("round", "")})
         return dict(
             stage="beta",
             service_meta_author="Department for Levelling up Housing and Communities",
-            accessibility_statement_url=Config.APPLICANT_FRONTEND_ACCESSIBILITY_STATEMENT_URL,  # noqa
-            cookie_policy_url=Config.APPLICANT_FRONTEND_COOKIE_POLICY_URL,
-            contact_us_url=Config.APPLICANT_FRONTEND_CONTACT_US_URL
-            + f"?fund={request.args.get('fund', '')}&round={request.args.get('round', '')}",
```
[the remark this answers is not in the document]
Comment (gidsg) [reply app.py:123]: @sfount this logic (whatever it is doing) has been preserved, it's just deduplicated now in: https://github.com/communitiesuk/funding-service-design-authenticator/blob/0db9a4ed6925c727075121146bde90b0854e6549/app.py#L119

**Review by hamzabinkhalid — APPROVED** [review]
LGTM

**Pull request merged**